### ISSUE TITLE
TypeError: 'int' object is not subscriptable

### ISSUE THREAD

**Issue by everythingnew** [SC4RECOIN/LSTM-Crypto-Price-Prediction#3]
Hi, thank you for your helpful repo. I tried to run lstm.py and getting following Errors:

```
Using TensorFlow backend.
2019-12-17 15:43:51.289029: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library cudart64_100.dll
C:\Envs\bml\lib\site-packages\sklearn\externals\joblib\__init__.py:15: FutureWarning:

sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.

Traceback (most recent call last):
  File "C:/projects/crypto2/lstm.py", line 118, in <module>
    X, y = extract_data()
  File "C:/projects/crypto2/lstm.py", line 24, in extract_data
    macd = Macd(6, 12, 3).values
  File "C:\projects\crypto2\technical_analysis\macd.py", line 37, in __init__
    self.values = self.calc_macd(data)
  File "C:\projects\crypto2\technical_analysis\macd.py", line 41, in calc_macd
    short_ema = self.short_ema.calc_ema(data)
  File "C:\projects\crypto2\technical_analysis\macd.py", line 12, in calc_ema
    seed = sum(data[:self.period]) / self.period
TypeError: 'int' object is not subscriptable
```

i cant really understand why am i getting this error. could you help me with that?

**Comment by SC4RECOIN** [comment]
Hey everythingnew,
It's been a long time since I worked on this code but I can try to help you out. I must been messing around with the data and technical indicator constructors last time I was working with this. You're getting this error because the indicators are expecting ```data``` as the first argument but are getting int.
```python
class Macd(object):
    def __init__(self, data, short_pd=12, long_pd=26, sig_pd=9):
```
I was loading the data inside the constructor rather than passing which caused this conflict. You'll need to correct this issue. I might look into it later if I have time

**Comment by everythingnew** [comment]
Thank you very much for Answer. i just added 
`data = np.load('historical_data/hist_data.npy')`
to the code and should be passed through Macd,Stoch_Rsi,Coppock and its worked

**Comment by SC4RECOIN** [comment]
Yes sorry fixed in ```a20ca17896058f6844b9c18722b3254c0efab9f0```.
Thanks for catching that
